packages/jest-runtime/src/index.ts

```diff
@@ -1684,7 +1684,7 @@ export default class Runtime {
     const moduleLookup: Record<string, VMModule> = {};
     for (const {module} of imports) {
       if (moduleLookup[module] === undefined) {
-        moduleLookup[module] = await this.loadEsmModule(
+        moduleLookup[module] = await this.linkAndEvaluateModule(
```

Comment:
It's still added to the cache, right?

Reply:
I don’t know for sure unfortunately. There are calls to `this._esmModuleLinkingMap` → `get/set` inside `linkAndEvaluateModule` but I don’t know what that means. All I know is that `loadEsmModule` cannot be used here – tests don’t pass.

Reply:
We need to make sure the module is checked for in `_esModuleRegister` or what it's called. And then re-user or added afterwards

Reply:
AFAIU, the module already ends up in `_esModuleRegister` in `wasm-bindgen` libraries. We have this import flow: 

```
user-code.js
↓
[bindgen package] / index.js
↓
[bindgen package] / index_bg.js
↓↑
[bindgen package] / index_bg.wasm
```

Because `index_bg.js` is imported from `index_bg.js`, `index_bg.js` ends up in the cache registry by then. But because of circular deps, calling `loadEsmModule` instead of `linkAndEvaluateModule` fails because it returns an object with `status: 'linking'` instead of a string path.

It’s possible to imagine a Wasm file that imports from some other JS file. I don’t know what will happen in this case TBH. Perhaps we can focus on fixing a practical problem in this PR and leave this theoretical scenario for later. Wasm support is still experimental anyway.

Feel free to push changes if you can think of any! The key added value of this PR is a new test case, not the fix.